### DISCUSSION TITLE
Symfony Version 2 to Symfony version 7 upgrade..How can i do that?

### DISCUSSION THREAD

**Discussion by haritaRaba** [symfony/symfony#57423] [Q&A]
I have one project which is in symfony version 2 and now i want to upgrade it to version 7 ....how can i easily do this as a symfony developer?
what challanges i can face in order to achieve this?
Please suggest me easiest way.
Thank you.

**Comment by chapterjason** [comment]
I would say step by step and have a look at reactor. (https://github.com/rectorphp/rector)

**Comment by GromNaN** [comment]
So many things have changed between versions 2 and 7. Trying to upgrade 6 versions on a project will be a pain. I don't recommend upgrading to every major version either, as things were introduced, improved, deprecated and removed over time.

You can create a new Symfony 7 project and migrate each feature one by one by copying and refactoring the corresponding files. If you want to do a gradual migration, you can deploy the old and new applications at the same time and use nginx to route to one or other of the applications depending on the url (if this is possible for your application).

**Comment by alexandre-le-borgne** [comment]
I'll start a new project in Symfony 7 and move the files 1 by 1 (deleting them from the old version once migrated) and adapting the code for the new version by making the minimum changes. Whatever happens, this task will be long and tedious. 

Once that's done, I'll use Rector to make the big, boring changes (moving annotations to attributes, etc.).

If there are tests, run them and correct them until they pass (making sure they already passed on the old version).

**Reply by alexandre-le-borgne** [reply]
@haritaRaba So, did it go well?

**Comment by ghost** [comment]
only complete rewrite
